### ISSUE TITLE
Azure App Service deploy v4 fails with 409 error when Deployment Type is set to Run From Package

### ISSUE THREAD

**Issue by dpolivy** [microsoft/azure-pipelines-tasks#12671]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: AzureRmWebAppDeployment

## Environment
- Server - Azure Pipelines
    
    - Account/Project: CellarTracker

- Agent - Hosted
    
## Issue Description

To speed up my deployments, Support has advised me I should use Run From Package instead of the standard Zip Deploy method. The task is defaulting to zip deploy because I have a post-deployment script that copies the applicationHost.xdt transform from my zip package to the proper place on the App Service.

When I manually change the deployment method to `Run From Package` in the task, it fails as the task is attempting to `PUT site/wwwroot/app_offline.htm`, but since we're running from package, `wwwroot` is read-only and the call returns a 409. This, the deployment fails. Here is the specific error returned:

````
##[error]Error: Unable to run the script on Kudu Service. Error: Error: Failed to upload file 'site/wwwroot/app_offline.htm from Kudu. Error: Conflict (CODE: 409)
````

I believe this is some kind of bug, where the task should explicitly be setting the `TakeAppOfflineFlag` to `false` on the task when Run From Package is explicitly chosen as the deployment method.

My pipeline is created via the UI as a Release pipeline, so I have no option to manually modify the YAML for the task to set this myself. Attempting to change the Deployment Method to Zip Deploy exposes the setting in the UI, but unchecking it and then changing the deployment method back to Run From Package does not resolve the problem.

### Task logs

I can provide logs privately, or, additionally, they have been provided to support on my existing ticket.

**Comment by aksm-ms** [comment]
As per current design, "TakeAppOfflineFlag" is not available for edit on the task for the deployment method "Run from package".

**Comment by dpolivy** [comment]
@aksm-ms Yes, that much is obvious. However, the value seems to default to `true`, and it is very clear that if the task attempts to take the app offline when the app service is already set to run from package, then it will fail with 409.

This was pretty easy for me to reproduce by creating a new pipeline with the task, and changing the deployment type to Run From Package.

**Comment by dpolivy** [comment]
Looks like this is a bug that is fixed by #12734. Thanks for fixing it!
